### PR TITLE
Add kwargs for new SurfaceFluxes release

### PR DESCRIPTION
**Pull request by costachris**
## Purpose 
Add kwargs for new SurfaceFluxes release, update `weighted_jabobian` calls.

### CODE REVIEW

File: src/Fields.jl [hunk 1]
```diff
@@ -68,13 +68,13 @@ function FieldFromNamedTuple(space, initial_conditions::Function, ::Type{FT}, pa
 end
 
 function Base.cumsum(field::CC.Fields.FiniteDifferenceField)
-    Base.cumsum(parent(CC.Fields.weighted_jacobian(field)) .* vec(field), dims = 1)
+    Base.cumsum(parent(CC.Spaces.weighted_jacobian(field)) .* vec(field), dims = 1)
```
Comment (charleskawczynski) [inline src/Fields.jl:71]:
```suggestion
    Base.cumsum(parent(CC.Spaces.weighted_jacobian(axes(field))) .* vec(field), dims = 1)
```

**Comment by charleskawczynski** [comment]
@costachris, I suspect that some small behavioral changes are expected due to changes in Thermodynamics / SurfaceFluxes

**Comment by costachris** [comment]
> @costachris, I suspect that some small behavioral changes are expected due to changes in Thermodynamics / SurfaceFluxes

yep, I'll update the MSE tables.

**Comment by costachris** [comment]
@charleskawczynski I had to update the CI a bit to get the macOS build working again. 1) The older version of ClimaParameters allowed in the test env was breaking, 2) the build was using julia 1.8.3 (which isn't yet available on HPC), so I pinned that to 1.8.2

**Pull request merged**